lib/create-environment-api.ts

```diff
@@ -159,6 +167,218 @@ export default function createEnvironmentApi(makeRequest: MakeRequest) {
+     * @param bulkActionId - ID of the BulkAction to fetch
+     * @returns - Promise with the BulkAction
+     *
+     * @example ```javascript
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getSpace('<space_id>')
+     * .then((space) => space.getEnvironment('<environment-id>'))
+     * .then((environment) => environment.getBulkAction('<bulk_action_id>'))
+     * .then((bulkAction) => console.log(bulkAction))
+     * ```
+     */
+    getBulkAction<T extends BulkActionPayload = any>(bulkActionId: string): Promise<BulkAction<T>> {
+      const raw = this.toPlainObject() // Environment object
+
+      return makeRequest({
+        entityType: 'BulkAction',
+        action: 'get',
+        params: {
+          spaceId: raw.sys.space.sys.id,
+          environmentId: raw.sys.id,
+          bulkActionId,
+        },
+      }).then((data) => wrapBulkAction<T>(makeRequest, data))
+    },
+
+    /**
+     * @description Creates a BulkAction that will attempt to publish all items contained in the payload.
+     * See: https://www.contentful.com/developers/docs/references/content-management-api/#/reference/bulk-actions/publish-bulk-action
+     * @param {BulkActionPayload} payload - Object containing the items to be processed in the bulkAction
+     * @returns - Promise with the BulkAction
+     *
+     * @example
+     *
+     * ```javascript
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * const payload = {
+     *  entities: {
+     *    sys: { type: 'Array' }
+     *    items: [
+     *      { sys: { type: 'Link', id: '<entry-id>', linkType: 'Entry', version: 2 } }
+     *    ]
+     *  }
+     * }
+     *
+     * // Using Thenables
+     * client.getSpace('<space_id>')
+     * .then((space) => space.getEnvironment('<environment-id>'))
+     * .then((environment) => environment.createPublishBulkAction(payload))
+     * .then((bulkAction) => console.log(bulkAction.waitProcessing()))
+     * .catch(console.error)
+     *
+     * // Using async/await
+     * try {
+     *  const space = await client.getSpace('<space_id>')
+     *  const environment = await space.getEnvironment('<environment_id>')
+     *  const bulkActionInProgress = await environment.createPublishBulkAction(payload)
+     *
+     *  // You can wait for a recently created BulkAction to be processed by using `bulkAction.waitProcessing()`
+     *  const bulkActionCompleted = await bulkActionInProgress.waitProcessing()
+     *  console.log(bulkActionCompleted)
+     * } catch (error) {
+     *  console.log(error)
+     * }
+     * ```
+     */
+    createPublishBulkAction(payload: BulkActionPayload) {
```

Comment:
Does the payload need to be a `BulkActionPublishPayload` here?

Reply:
oh, you are correct. I fixed the return but didn't update this one. Fixed 👍🏽